### PR TITLE
Fixes forloop issue in nested component

### PR DESCRIPTION
**Pull request by suresh-gangumalla**

### CODE REVIEW

**Review by wouterlucas — APPROVED** [review]
LGTM just some curiosities

File: src/helpers/deepEqualArray.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const deepEqualArray = (array1, array2) => {
+  if (array1 === array2) return true
+  if (array1.length !== array2.length) return false
```
[the remark this answers is not in the document]
Comment (michielvandergeest) [reply src/helpers/deepEqualArray.js:20]: L19 checks if the arrays point to the same reference. I would think that to be a cheaper operation than retrieving and comparing the length?

File: src/helpers/deepEqualArray.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const deepEqualArray = (array1, array2) => {
+  if (array1 === array2) return true
```
[the remark this answers is not in the document]
Comment (michielvandergeest) [reply src/helpers/deepEqualArray.js:19]: Well, in order to trigger the reactivity we need to 'reassign' the array. So in practice, from the reactivity side we wouldn't fall into this condition as we're never comparing references. It's here mostly for completeness and reusability of the function (with the previous assumption that a reference check is cheap).

File: src/helpers/deepEqualArray.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (wouterlucas) [reply src/helpers/deepEqualArray.js:20]: ah right yes, if they're both from the same ref. I see what you mean, but if they're not wouldn't that trigger a more thorough compare?

[the remark this answers is not in the document]
Comment (michielvandergeest) [reply src/helpers/deepEqualArray.js:20]: the logic is the following order
- same ref? then they are deeply equal, period.
- not the same length? then they can't be deeply equal
- not one of the above? then let's go inspect the array further until we can be sure they are equal (or not)

[the remark this answers is not in the document]
Comment (wouterlucas) [reply src/helpers/deepEqualArray.js:20]: okay 😸

**Pull request merged**